### PR TITLE
Improve README

### PR DESCRIPTION
**Pull request by kmaehashi**
Improved tag line and description, add example snippet to README.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,8 +17,21 @@
 | [**API Reference**](https://docs.cupy.dev/en/stable/reference/)
 | [**Forum**](https://groups.google.com/forum/#!forum/cupy)
 
-*CuPy* is an implementation of NumPy-compatible multi-dimensional array on CUDA.
-CuPy consists of the core multi-dimensional array class, `cupy.ndarray`, and [many functions](https://docs.cupy.dev/en/stable/reference/comparison.html) on it.
+CuPy is a NumPy/SciPy-compatible array library for GPU-accelerated computing with Python.
+CuPy acts as a [drop-in replacement](https://docs.cupy.dev/en/stable/reference/comparison.html) for existing NumPy/SciPy code to make it runnable on NVIDIA CUDA or AMD ROCm platform.
```
Comment (emcastillo) [inline README.md:21]:
```suggestion
CuPy acts as a [drop-in replacement](https://docs.cupy.dev/en/stable/reference/comparison.html) to run existing NumPy/SciPy code on NVIDIA CUDA or AMD ROCm platforms.
```

File: README.md [hunk 2]
```diff
@@ -17,8 +17,21 @@
...
+>>> x = cp.arange(6).reshape(2, 3).astype('f')
+>>> x
+array([[ 0.,  1.,  2.],
+       [ 3.,  4.,  5.]], dtype=float32)
+>>> x.sum(axis=1)
+array([  3.,  12.], dtype=float32)
+```
+
+CuPy also provides access to low-level CUDA features.
+You can pass `ndarray` to existing CUDA C/C++ program via [RawKernels](https://docs.cupy.dev/en/stable/user_guide/kernel.html#raw-kernels), use [Streams](https://docs.cupy.dev/en/stable/reference/cuda.html) for performance, or even call [CUDA Runtime APIs](https://docs.cupy.dev/en/stable/reference/cuda.html#runtime-api) directly.
```
Comment (emcastillo) [inline README.md:34]:
```suggestion
You can pass `ndarray` to existing CUDA C/C++ programs via [RawKernels](https://docs.cupy.dev/en/stable/user_guide/kernel.html#raw-kernels), use [Streams](https://docs.cupy.dev/en/stable/reference/cuda.html) for performance, or even call [CUDA Runtime APIs](https://docs.cupy.dev/en/stable/reference/cuda.html#runtime-api) directly.
```

**Comment by kmaehashi** [comment]
Thanks, applied suggestions!

**Comment by emcastillo** [comment]
Merging as it does not require CIs

**Review by emcastillo — APPROVED** [review]

**Pull request merged**